backend-base/src/main/java/org/eclipse/che/incubator/workspace/telemetry/base/AbstractAnalyticsManager.java

```diff
@@ -25,45 +43,72 @@
-
-import static java.lang.Long.parseLong;
-import static org.eclipse.che.multiuser.machine.authentication.shared.Constants.USER_ID_CLAIM;
-import static org.slf4j.LoggerFactory.getLogger;
+import io.jsonwebtoken.JwtParser;
+import io.jsonwebtoken.Jwts;
 
 public abstract class AbstractAnalyticsManager {
   private static final Logger LOG = getLogger(AbstractAnalyticsManager.class);
 
   protected final String workspaceId;
   protected String userId = "";
 
-  @VisibleForTesting final protected String workspaceName;
-  @VisibleForTesting final protected String factoryId;
-  @VisibleForTesting final protected String stackId;
-  @VisibleForTesting final protected String factoryName;
-  @VisibleForTesting final protected String factoryOwner;
-  @VisibleForTesting final protected String factoryUrl;
-  @VisibleForTesting final protected String createdOn;
-  @VisibleForTesting final protected String updatedOn;
-  @VisibleForTesting final protected String stoppedOn;
-  @VisibleForTesting final protected String stoppedAbnormally;
-  @VisibleForTesting final protected String lastErrorMessage;
-  @VisibleForTesting final protected String osioSpaceId;
-  @VisibleForTesting final protected String sourceTypes;
-  @VisibleForTesting final protected String startNumber;
-  @VisibleForTesting final protected List<String> pluginNames;
-
-  @VisibleForTesting final protected Long age;
-  @VisibleForTesting final protected Long returnDelay;
-  @VisibleForTesting final protected Boolean firstStart;
+  @VisibleForTesting
+  protected long noActivityTimeout = 60000 * 3;
+
+  @VisibleForTesting
+  final protected String workspaceName;
+  @VisibleForTesting
+  final protected String factoryId;
+  @VisibleForTesting
+  final protected String stackId;
+  @VisibleForTesting
+  protected String factoryName;
+  @VisibleForTesting
+  protected String factoryOwner;
+  @VisibleForTesting
+  protected String factoryUrl;
+  @VisibleForTesting
+  final protected String createdOn;
+  @VisibleForTesting
+  final protected String updatedOn;
+  @VisibleForTesting
+  final protected String stoppedOn;
+  @VisibleForTesting
+  final protected String stoppedAbnormally;
+  @VisibleForTesting
+  final protected String lastErrorMessage;
+  @VisibleForTesting
+  final protected String osioSpaceId;
+  @VisibleForTesting
+  final protected String sourceTypes;
+  @VisibleForTesting
+  final protected String startNumber;
+  @VisibleForTesting
+  final protected List<String> pluginNames;
+
+  @VisibleForTesting
+  final protected Long age;
+  @VisibleForTesting
+  final protected Long returnDelay;
+  @VisibleForTesting
+  final protected Boolean firstStart;
+
+  @VisibleForTesting
+  protected String workspaceStartingUserId = null;
+
+  @VisibleForTesting
+  protected Map<String, Object> commonProperties;
+
+  @VisibleForTesting
+  protected static long pingTimeoutSeconds = 30;
+
+  @VisibleForTesting
+  protected static long pingTimeout = pingTimeoutSeconds * 1000;
```

Comment:
why not constants?

Reply:
I think a subclass might want to change this